descriptor/qsarsubstance/src/main/java/org/openscience/cdk/qsar/descriptors/substance/OxygenAtomCountDescriptor.java

```diff
@@ -0,0 +1,120 @@
+ * along with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+package org.openscience.cdk.qsar.descriptors.substance;
+
+import org.openscience.cdk.exception.CDKException;
+import org.openscience.cdk.interfaces.IChemObjectBuilder;
+import org.openscience.cdk.interfaces.IIsotope;
+import org.openscience.cdk.interfaces.IMolecularFormula;
+import org.openscience.cdk.interfaces.ISubstance;
+import org.openscience.cdk.qsar.DescriptorSpecification;
+import org.openscience.cdk.qsar.DescriptorValue;
+import org.openscience.cdk.qsar.result.IDescriptorResult;
+import org.openscience.cdk.qsar.result.IntegerResult;
+import org.openscience.cdk.qsar.result.IntegerResultType;
+
+/**
+ * Descriptor that returns the number of oxygens in the chemical
+ * formula. Originally aimed at metal oxide nanoparticles.
+ *
+ * @author      egonw
+ * @cdk.githash
+ */
+public class OxygenAtomCountDescriptor implements ISubstanceDescriptor {
+
+	/** {@inheritDoc} */ @Override
+	public String[] getDescriptorNames() {
+        return new String[]{"NoMe"};
+	}
+
+	/** {@inheritDoc} */ @Override
+	public String[] getParameterNames() {
+        return new String[0];
+	}
+
+	/** {@inheritDoc} */ @Override
+	public Object getParameterType(String substance) {
+		return null;
+	}
+
+	/** {@inheritDoc} */ @Override
+	public Object[] getParameters() {
+		return new Object[0];
+	}
+
+	/** {@inheritDoc} */ @Override
+	public DescriptorSpecification getSpecification() {
+	    return new DescriptorSpecification(
+	        "http://egonw.github.com/resource/NM_001002",
+	        this.getClass().getName(),
+	        "The Chemistry Development Kit"
+	    );
+	}
+
+	/** {@inheritDoc} */ @Override
+	public void setParameters(Object[] parameters) throws CDKException {
+		return; // no parameters
+	}
+
+	/** {@inheritDoc} */ @Override
+	public DescriptorValue calculate(ISubstance substance) {
+        if (substance == null) return newNaNDescriptor();
+
+	    IMolecularFormula molFormula = SubstanceManipulator.getChemicalComposition(substance);
+	    if (molFormula == null) return newNaNDescriptor();
+
+	    int count = 0;
+        for (IIsotope isotope : molFormula.isotopes()) {
+            if ("O".equals(isotope.getSymbol())) {
+                count = molFormula.getIsotopeCount(isotope);
+                return new DescriptorValue(
+                    getSpecification(),
+                    getParameterNames(),
+                    getParameters(),
+                    new IntegerResult(count),
+                    getDescriptorNames()
+                );
+            }
+        }
+	    
+		return newNaNDescriptor();
+	}
+
+	/**
+	 * Internal method to return a NaN value.
+	 */
+	private DescriptorValue newNaNDescriptor() {
+	    return new DescriptorValue(
+	        getSpecification(),
+	        getParameterNames(),
+	        getParameters(),
+	        new IntegerResult((int)Double.NaN),
```

Comment:
Oh, got it... yeah, this isn't right... but, a lot of existing code with integer return values is affected :/